src/realm/object-store/sync/sync_session.cpp

```diff
@@ -1323,3 +1323,13 @@ void SyncSession::ConnectionChangeNotifier::invoke_callbacks(ConnectionState old
     }
     m_callback_index = npos;
 }
+
+util::Future<std::string> SyncSession::send_test_command(std::string body)
+{
+    util::CheckedLockGuard lk(m_state_mutex);
+    if (!m_session) {
+        return Status{ErrorCodes::RuntimeError, "Session doesn't exist to send test command on"};
+    }
+
+    return m_session->send_test_command(std::move(body));
```

Comment:
Should there be a check for an empty body before trying to send it?

Reply:
I added a bunch of validation deeper down in the callstack for this.